src/sidebar/services/frame-sync.js

```diff
@@ -181,17 +226,28 @@ export class FrameSyncService {
   _connectGuest(port) {
     /** @type {PortRPC<GuestToSidebarEvent, SidebarToGuestEvent>} */
     const guestRPC = new PortRPC();
-    this._guestRPC.push(guestRPC);
 
-    /** @type {string|null} */
-    let frameIdentifier;
+    // Add guest RPC to map with a temporary ID until we learn the real ID.
+    //
+    // We need to add the guest to the map immediately so that any notifications
+    // sent from this service to all guests, before we learn the real frame ID,
+    // are sent to this new guest.
+    ++this._nextGuestId;
+    let frameIdentifier = /** @type {string|null} */ (
+      `temp-${this._nextGuestId}`
+    );
+    this._guestRPC.set(frameIdentifier, guestRPC);
```

Comment:
In this PR I have not changed how frame IDs are allocated. If we change this in future so that frame IDs are either allocated entirely within the sidebar or in such a way that we can know them as soon as the guest connects, then we can simplify this code.

Reply:
> In the future, we could easily add a capability to PortFinder#discover to send a frame identifier.

Yes. That's definitely one possibility.